File: app/src/main/java/org/metabrainz/mobile/api/data/search/entity/Label.java
package org.metabrainz.mobile.api.data.search.entity;

import com.google.gson.annotations.SerializedName;

public class Label {
    @SerializedName("id")
    private String mbid;
    private String name;
    private String disambiguation;
    private String type;
    @SerializedName("life-span")
    private LifeSpan lifeSpan;
    private String country;
    private Area area;

    public String getDisambiguation() {
        return disambiguation;
    }

    public void setDisambiguation(String disambiguation) {
        this.disambiguation = disambiguation;
    }

    @Override
    public String toString() {
        return "Label{" +
                "mbid='" + mbid + '\'' +
                ", name='" + name + '\'' +
                ", disambiguation='" + disambiguation + '\'' +
                ", lifeSpan=" + lifeSpan +
                ", type='" + type + '\'' +
                ", country='" + country + '\'' +
                ", area=" + area +
                '}';
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getMbid() {
        return mbid;
    }

    public void setMbid(String mbid) {
        this.mbid = mbid;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getdisambiguation() {
        return disambiguation;
    }

    public void setdisambiguation(String disambiguation) {
        this.disambiguation = disambiguation;
    }

    public LifeSpan getLifeSpan() {
        return lifeSpan;
    }

    public void setLifeSpan(LifeSpan lifeSpan) {
        this.lifeSpan = lifeSpan;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public Area getArea() {
        return area;
    }

    public void setArea(Area area) {
        this.area = area;
    }
}

File: app/src/main/java/org/metabrainz/android/model/mbentity/Label.kt
package org.metabrainz.android.model.mbentity

import com.google.gson.annotations.SerializedName
import org.metabrainz.android.model.entities.LifeSpan
import org.metabrainz.android.model.entities.Link
import java.util.*

class Label : org.metabrainz.android.model.mbentity.MBEntity() {
    var name: String? = null
    var type: String? = null

    @SerializedName("label-code")
    var code: String? = null
        get() = if (field != null) "LC $field" else ""

    @SerializedName("life-span")
    var lifeSpan: LifeSpan? = null
    var country: String? = null
    var area: org.metabrainz.android.model.mbentity.Area? = null
    var relations: MutableList<Link> = ArrayList()
        set(relations) {
            this.relations.addAll(relations)
        }
    var releases: MutableList<org.metabrainz.android.model.mbentity.Release> = ArrayList()
        set(releases) {
            this.releases.addAll(releases)
        }

    override fun toString(): String {
        return "Label{" +
                "mbid='" + mbid + '\'' +
                ", name='" + name + '\'' +
                ", disambiguation='" + disambiguation + '\'' +
                ", lifeSpan=" + lifeSpan +
                ", type='" + type + '\'' +
                ", country='" + country + '\'' +
                ", area=" + area +
                '}'
    }
}